cogs/highlight.py

```diff
@@ -198,8 +209,18 @@ def _is_registered(self, guild_idx, guild_id, user_id):
             notify_msg += "[{0}] {1.author.name}#{1.author.discriminator}: {1.content}\n".format(time,msg)
         await self.bot.send_message(user,notify_msg)
         
-    async def _last_spoken(self):
-        pass # TODO: check if user sent message in last 10 seconds or so, if so dont send a PM
+    async def _is_active(self, user_id, channel, message):
+        # NOTE: this is a naive approach to checking activity, for now to keep simple, just see if user
+        # created a message in the last 50 messages. and if they did, was in less than 20 seconds since 
+        # the message we are currently checking for highlight words

```

Comment:
Hmm, you could use the ``on_message()`` event listener to update a dictionary of some sort to keep track of when the user last sent a message?  I wouldn't keep it in a file per say, just in RAM would suffice.  Your thoughts?